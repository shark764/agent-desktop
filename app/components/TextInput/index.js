/**
*
* TextInput
*
*/

import React, { PropTypes } from 'react';

import { injectIntl, intlShape } from 'react-intl';

import Radium from 'radium';


function TextInput(props) {
  const { formatMessage } = props.intl;
  const styles = {
    base: {
      width: '282px',
      height: '44px',
      backgroundColor: '#ffffff',
      border: 'solid 1px #979797',
      fontSize: '16px',
      fontWeight: 'light',
      fontStyle: 'normal',
      fontStretch: 'normal',
      color: '#494949',
      outline: 'none',
      padding: '11px',
      borderRadius: '2px',

      ':focus': {
        boxShadow: '0 0 6px 1px rgba(0, 0, 0, 0.12)',
        border: 'solid 1px #23CEF5',
      },
    },
  };

  return (
    <input
      style={[styles.base, styles.center, props.style]}
      type={props.type || 'text'}
      key={formatMessage(props.placeholder)}
      value={props.value}
      placeholder={formatMessage(props.placeholder)}
      onChange={(e) => props.cb(e.target.value)}
      autoComplete={props.autocomplete || 'on'}
      onKeyUp={props.onKeyUp || ''}
      autoFocus={props.autoFocus}
    />
  );
}

TextInput.propTypes = {
  intl: intlShape.isRequired,
  placeholder: PropTypes.object.isRequired,
  cb: PropTypes.func.isRequired,
  autocomplete: PropTypes.string,
  value: PropTypes.string.isRequired,
  style: PropTypes.object,
  type: PropTypes.string,
  onKeyUp: PropTypes.func,
  autoFocus: PropTypes.bool,
};

export default injectIntl(Radium(TextInput));
