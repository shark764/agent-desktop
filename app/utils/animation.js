/**
 * throttle max function call rate to browser refresh rate
 */

const requestAnimationFrame =
  window.requestAnimationFrame ||
  window.webkitRequestAnimationFrame ||
  window.mozRequestAnimationFrame ||
  ((callback) => window.setTimeout(callback, 1000 / 60));

function throttler(...callbackArgs) {
  if (!this.running) {
    this.running = true;
    requestAnimationFrame(() => {
      this.callback(...callbackArgs);
      this.running = false;
    });
  }
}

export function throttleDecorator(funcToThrottle) {
  return throttler.bind({ callback: funcToThrottle, running: false });
}
