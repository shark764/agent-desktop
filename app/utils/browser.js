/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

// Check if browser used is IE11 or greater
// Returns undefined for Chrome|Firefox|Edge
export const isIeEleven = () => document.documentMode && document.documentMode >= 11;
