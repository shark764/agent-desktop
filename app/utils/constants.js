/*
 * Copyright Â© 2015-2020 Serenova, LLC. All rights reserved.
 */

export const colors = {
  lightGray: '#ebebeb',
  offWhite: '#e4e4e4',
  lightRed: '#FE4565',
  regularRed: '#E43D5A',
  darkRed: '#CB3750',
  darkGreen: '#001b1e',
  hoverDarkGreen: 'rgba(0, 27, 30, 0.85)',
  black: '#000',
  white: '#FFFFFF',
  skylightBlue: '#23cdf4',
  toggleChecked: '#54B84F',
  toggleCheckedDisabled: '#b7e3b3',
};
