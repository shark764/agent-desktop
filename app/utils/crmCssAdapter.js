/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
  * crmCssAdapter
  *
  * Updates CSS as needed to work with various CRM integrations
  *
  */

export default function crmCssAdapter (styleObject, styleNameArray, crmModule) {
  let styleAffix;
  if (crmModule === 'salesforce-lightning') {
    styleAffix = 'SfLightning';
  }

  styleNameArray.forEach((styleName) => {
    Object.assign(
      styleObject[styleName],
      styleObject[`${styleName}${styleAffix}`]
    );
  });
}