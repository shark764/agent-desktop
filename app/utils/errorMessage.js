import errorMessages from 'containers/Errors/messages';

export const generateErrorMessage = (errorObject, formatMessage) => {
  let errorDescriptionMessage;
  const { code, interactionFatal } = errorObject;
  if (interactionFatal) {
    errorDescriptionMessage = formatMessage(errorMessages.interactionFailed);
  } else if (code && errorMessages[code]) {
    // Specific error message is found for this code
    errorDescriptionMessage = formatMessage(errorMessages[code]);
  } else if (errorObject.message) {
    // Fallback to message provided by error (for SDK errors)
    errorDescriptionMessage = errorObject.message;
  } else {
    // Use default for other errors (probably made by us)
    errorDescriptionMessage = errorMessages.default;
  }
  if (code) {
    errorDescriptionMessage += formatMessage(errorMessages.code, { code });
  }
  if (errorObject.data && errorObject.data.errorDescription) {
    errorDescriptionMessage += formatMessage(errorMessages.errorDescription, {
      errorDescription: errorObject.data.errorDescription,
    });
  }
  return errorDescriptionMessage;
};
