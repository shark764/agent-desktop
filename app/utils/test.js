import { IntlProvider } from 'react-intl';
import enUS from 'translations/en-US.json';

export const getIntlContext = () => {
  // Construct a new `IntlProvider` instance by passing `props` and
  // `context` as React would, then call `getChildContext()` to get the
  // React Intl API, complete with the `format*()` functions.
  const intlProvider = new IntlProvider({ locale: 'en-US', messages: enUS }, {});
  const { intl } = intlProvider.getChildContext();
  return intl;
};

export default { getIntlContext };
