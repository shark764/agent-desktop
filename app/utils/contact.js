/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import { isValidEmail, isPossibleNumber } from 'utils/validator';
import isURL from 'validator/lib/isURL';

export const formatValue = (attributeToValidate, value) => {
  let formattedValue;
  switch (attributeToValidate.type) {
    case 'phone':
      formattedValue = value.replace(/[^0-9+*#]/g, '');
      if (formattedValue.indexOf('+') !== 0 && formattedValue.length > 0) {
        formattedValue = `+${formattedValue}`;
      }
      break;
    case 'boolean':
      if (value === 'false' || value === '') formattedValue = false;
      else formattedValue = !!value;
      break;
    default:
      formattedValue = value;
  }
  return formattedValue;
};

export const getError = (attributeToValidate, value) => {
  let error = false;
  if (
    attributeToValidate.mandatory &&
    attributeToValidate.type !== 'boolean' &&
    (value === undefined || value.trim().length < 1)
  ) {
    error = 'errorRequired';
  } else if (value !== undefined && value.length) {
    switch (attributeToValidate.type) {
      case 'email':
        if (!isValidEmail(value)) {
          error = 'errorEmail';
        }
        break;
      case 'phone':
        try {
          if (!isPossibleNumber(value)) {
            error = 'errorPhone';
          }
        } catch (e) {
          error = 'errorPhone';
        }
        break;
      case 'link':
        if (
          !isURL(value, {
            protocols: ['http', 'https'],
            require_protocol: true,
          })
        ) {
          error = 'errorLink';
        }
        break;
      case 'number':
        if (Number.isNaN(Number(value))) {
          error = 'errorNumber';
        }
        break;
      default:
        break;
    }
  }
  return error;
};

export const getLocaleLabel = (attribute, locale) => {
  if (attribute.label[locale] !== undefined) {
    return attribute.label[locale];
  } else if (attribute.label.all !== undefined) {
    // Special case for 'All'. We have our own local translation for it we use.
    return attribute.label.all;
  } else if (attribute.objectName !== undefined) {
    // Fallback for attributes
    return attribute.objectName;
  } else {
    // Fallback for section headers
    return '';
  }
};

export function createSearchQuery(query, nextPage) {
  const encodedQuery = {};
  Object.keys(query).forEach(queryName => {
    const queryToEncode = query[queryName];
    encodedQuery[queryName] = encodeURIComponent(queryToEncode);
  });
  return {
    query: Object.assign(encodedQuery, {
      page: nextPage !== undefined ? nextPage : 1,
    }),
  };
}
