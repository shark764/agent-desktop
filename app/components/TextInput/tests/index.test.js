/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';
import { getIntlContext } from 'utils/test';

import TextInput from '../index';

describe('<TextInput />', () => {
  it('should render correctly', () => {
    const rendered = shallow(
      <TextInput.WrappedComponent
        id="mockId"
        intl={getIntlContext()}
        cb={() => {}}
        value="mockValue"
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
