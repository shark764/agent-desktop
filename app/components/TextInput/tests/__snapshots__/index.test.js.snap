// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should render correctly 1`] = `
<input
  autoComplete="on"
  data-radium={true}
  id="mockId"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown=""
  onKeyUp={[Function]}
  placeholder=""
  style={
    Object {
      "backgroundColor": "#ffffff",
      "border": "solid 1px #979797",
      "borderBottom": "none",
      "borderLeft": "none",
      "borderRadius": "2px",
      "borderRight": "none",
      "borderTop": "none",
      "color": "#494949",
      "fontSize": "16px",
      "fontStretch": "normal",
      "fontStyle": "normal",
      "fontWeight": "light",
      "height": "44px",
      "outline": "none",
      "padding": "11px",
      "width": "282px",
    }
  }
  tabIndex={0}
  type="text"
  value="mockValue"
/>
`;
